src/adams_methods.jl

```diff
@@ -0,0 +1,655 @@
+            # equation (5.5) in Hairer et al, Vol 1, where we note that here
+            # y_np1 = y_{k}(t_{n+1})
+
+            y_np1 = y[n]
+            for j= 1:steporder
+              y_np1 += h*g[j]* ϕstar[2,j]
+            end
+            # TODO: figure out why the following does not work
+            # for j= 1:steporder
+            #     for i=1:length(y_np1)
+            #         y_np1[i] += h*g[j]*ϕstar[2,j][i]
+            #     end
+            # end
+
+            # Compute current estimate for derivative
+            dy_np1=F(t_np1,y_np1)
+            #TODO: figure out why the following doesn't work
+            #copy!(dy_np1,F(t_np1,y_np1))
+
+            #---################################################################
+            # Correction-Evaluaion (CE) Loop for implicit solver
+            #     See Harier p. 360 for a general description of this method
+            #---################################################################
+
+            # Set CE_loops to be how many correction-evaulation cycles
+            # desired. As least one loop is necessary, for the sake of the
+            # adaptive step selection
+            CE_loops = 1
+            loop_counter = 0
+
+            while loop_counter<CE_loops && n>=steporder+1
+                loop_counter += 1
+                # (C)orrection of y_np1
+                # Add new coefficients  based on this prediction
+                # computation of ϕ, ϕstar using recurrence relations
+                ϕ_coefs!(ϕ, ϕstar,dy_np1,1:steporder+1)
+
+                # Calculation of g_{steporder + 1}(n) coefficients using
+                # recurrence relations has already been done above, and does
+                # not depend on dy_np1 or y_np1, so no update necessary
+
+                # We correct the computation of y_np1 using y_np1
+                # as part of our prediction Harier equation 5.7
+
+                y_np1 = y_np1 + h*g[steporder+1]*ϕ[3,steporder+1]
+                # TODO: figure out why following does not work instead
+                # for i=1:length(y_np1)
+                #     y_np1[i] += h*g[steporder+1]*ϕstar[3,steporder+1][i]
+                # end
+
+                # (E)valuate estimate for next derivative again
+                dy_np1=F(t_np1,y_np1)
+            end
+            #---################################################################
+            # Error Check for step size `h` and Variable next step selection:
+            #     Check error to determine if step size should be accepted:
+            #     ->if yes, compute `next_h` for next step
+            #     ->if no, use this `next_h`` for next attempt at this step
+            #---################################################################
+
+            #Don't use adaptive steps until n >= steporder + 2
+            if n==1
+                #TODO: adaptive step selection for n<steporder +2
+                successful_step = true
+                next_h = h/4
```

Comment:
I don't understand this `next_h = h/4`, why reduce the step size for the second step?